### PR TITLE
r/fsx_windows_file_system - add support for multi-az

### LINKED ISSUE #12601: AWS FSx Subnet issue
_This issue was originally opened by @itdevon as hashicorp/terraform#24517. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._
    
<hr>
    
Having issues with AWS FSx, I have two subnets, but the error indicates only a max of 1 subnet. 

### Terraform Version

Terraform v0.11.14
+ provider.aws v2.55.0

### Terraform Configuration Files

resource "aws_fsx_windows_file_system" "orapreshopdbfs" {
  storage_capacity    = 2000
  subnet_ids          = ["subnet-1", "subnet-2"]
  throughput_capacity = 2048

  self_managed_active_directory { 
    dns_ips     = ["10.1.1.1", "10.1.1.2"]
    domain_name = "redacted"
    organizational_unit_distinguished_name = "redacted"
    password    = "redacted"
    username    = "redacted"
  }
}

### Debug Output

Error: aws_fsx_windows_file_system.orapreshopdbfs: subnet_ids: attribute supports 1 item maximum, config has 2 declared

### Expected Behavior

FSx is in mult-zone, so it will need to have two subnets. 

### Steps to Reproduce

Please list the full steps required to reproduce the issue, for example:
1. `terraform plan`

- hashicorp/terraform#6017

-->

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12601, #11246, #13527

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_fsx_windows_file_system: add support for multi-az
resource_aws_fsx_windows_file_system: add `SINGLE_AZ_2` deployment type
resource_aws_fsx_windows_file_system: adds `preferred_file_server_ip`, `remote_administration_endpoint` attributes
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (3134.55s)
--- PASS: TestAccAWSFsxWindowsFileSystem_multiAz (4520.65s)
--- PASS: TestAccAWSFsxWindowsFileSystem_singleAz2 (3354.19s)
```

### CODE REVIEW

**Comment by krzyszko** [comment]
@DrFaust92 I think preffered_subnet_id has to be provided explicitly when deployment_type

**Comment by JuanOI** [comment]
Hi @bflad  @DrFaust92 , any idea when this functionality will become available? This is currently blocking one of my customers. Thanks!

**Comment by adj97** [comment]
A nudge for documentation update ...

**Comment by DrFaust92** [comment]
rebase + fix lint

File: aws/resource_aws_fsx_windows_file_system.go [hunk 1]
```diff
@@ -174,6 +176,30 @@ func resourceAwsFsxWindowsFileSystem() *schema.Resource {
...
 			},
+			"deployment_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  fsx.WindowsDeploymentTypeSingleAz1,
+				ValidateFunc: validation.StringInSlice([]string{
+					fsx.WindowsDeploymentTypeMultiAz1,
+					fsx.WindowsDeploymentTypeSingleAz1,
+				}, false),
```
Comment (ewbankkit) [inline aws/resource_aws_fsx_windows_file_system.go:187]: Do you want to add `WindowsDeploymentTypeSingleAz2` as part of this PR? The option was added along with HDD support - https://github.com/terraform-providers/terraform-provider-aws/issues/12553 - but seems to have applicability  for SSD use cases.

Comment (DrFaust92) [reply aws/resource_aws_fsx_windows_file_system.go:187]: will add + test. it takes an hour to run and had to do a few tweaks 😢

Comment (DrFaust92) [reply aws/resource_aws_fsx_windows_file_system.go:188]: done.

**Comment by ewbankkit** [comment]
Verified applicable acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_basic -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_basic
=== PAUSE TestAccAWSFsxWindowsFileSystem_basic
=== CONT  TestAccAWSFsxWindowsFileSystem_basic
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (3274.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3274.464s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_singleAz2'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_singleAz2 -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_singleAz2
=== PAUSE TestAccAWSFsxWindowsFileSystem_singleAz2
=== CONT  TestAccAWSFsxWindowsFileSystem_singleAz2
--- PASS: TestAccAWSFsxWindowsFileSystem_singleAz2 (3404.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3404.884s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_multiAz'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_multiAz -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_multiAz
=== PAUSE TestAccAWSFsxWindowsFileSystem_multiAz
=== CONT  TestAccAWSFsxWindowsFileSystem_multiAz
--- PASS: TestAccAWSFsxWindowsFileSystem_multiAz (3936.95s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3937.003s

```

**Comment by ewbankkit** [comment]
Closes #13616.

File: website/docs/r/fsx_windows_file_system.html.markdown [hunk 2]
```diff
@@ -53,7 +53,7 @@ resource "aws_fsx_windows_file_system" "example" {
 The following arguments are supported:
 
 * `storage_capacity` - (Required) Storage capacity (GiB) of the file system. Minimum of 32 and maximum of 65536.
-* `subnet_ids` - (Required) A list of IDs for the subnets that the file system will be accessible from. File systems support only one subnet. The file server is also launched in that subnet's Availability Zone.
+* `subnet_ids` - (Required) A list of IDs for the subnets that the file system will be accessible from. to specify more than a single subnet set `deployment_type` to `MULTI_AZ_1`.
```
Comment (breathingdust) [inline website/docs/r/fsx_windows_file_system.html.markdown:56]:
```suggestion
* `subnet_ids` - (Required) A list of IDs for the subnets that the file system will be accessible from. To specify more than a single subnet set `deployment_type` to `MULTI_AZ_1`.
```

File: website/docs/r/fsx_windows_file_system.html.markdown [hunk 3]
```diff
@@ -65,6 +65,8 @@ The following arguments are supported:
 * `skip_final_backup` - (Optional) When enabled, will skip the default final backup taken when the file system is deleted. This configuration must be applied separately before attempting to delete the resource to have the desired behavior. Defaults to `false`.
 * `tags` - (Optional) A map of tags to assign to the file system.
 * `weekly_maintenance_start_time` - (Optional) The preferred start time (in `d:HH:MM` format) to perform weekly maintenance, in the UTC time zone.
+* `deployment_type` - (Optional) Specifies the file system deployment type, valid values are `MULTI_AZ_1` and `SINGLE_AZ_1`. Default values is `SINGLE_AZ_1`.
```
Comment (breathingdust) [inline website/docs/r/fsx_windows_file_system.html.markdown:68]:
```suggestion
* `deployment_type` - (Optional) Specifies the file system deployment type, valid values are `MULTI_AZ_1` and `SINGLE_AZ_1`. Default value is `SINGLE_AZ_1`.
```

File: website/docs/r/fsx_windows_file_system.html.markdown [hunk 4]
```diff
@@ -87,6 +89,8 @@ In addition to all arguments above, the following attributes are exported:
 * `network_interface_ids` - Set of Elastic Network Interface identifiers from which the file system is accessible.
 * `owner_id` - AWS account identifier that created the file system.
 * `vpc_id` - Identifier of the Virtual Private Cloud for the file system.
+* `preferred_file_server_ip` - the IP address of the primary, or preferred, file server.
```
Comment (breathingdust) [inline website/docs/r/fsx_windows_file_system.html.markdown:92]:
```suggestion
* `preferred_file_server_ip` - The IP address of the primary, or preferred, file server.
```

**Comment by breathingdust** [comment]
Verified applicable acceptance tests.

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_basic -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_basic
=== PAUSE TestAccAWSFsxWindowsFileSystem_basic
=== CONT  TestAccAWSFsxWindowsFileSystem_basic
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (2908.34s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2911.103s
```
```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_singleAz2'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_singleAz2 -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_singleAz2
=== PAUSE TestAccAWSFsxWindowsFileSystem_singleAz2
=== CONT  TestAccAWSFsxWindowsFileSystem_singleAz2
--- PASS: TestAccAWSFsxWindowsFileSystem_singleAz2 (3220.54s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3223.248s
```

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSFsxWindowsFileSystem_multiAz'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSFsxWindowsFileSystem_multiAz -timeout 120m
=== RUN   TestAccAWSFsxWindowsFileSystem_multiAz
=== PAUSE TestAccAWSFsxWindowsFileSystem_multiAz
=== CONT  TestAccAWSFsxWindowsFileSystem_multiAz
--- PASS: TestAccAWSFsxWindowsFileSystem_multiAz (3677.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3679.964s
```

**Pull request merged**